modules/apps/27-interchain-accounts/keeper/handshake.go

```diff
@@ -84,23 +89,37 @@ func (k Keeper) OnChanOpenTry(
 		return err
 	}
 
+	accAddr := types.GenerateAddress(counterparty.PortId)
+	parsedAddr := types.ParseAddressFromVersion(version)
+	if strings.Compare(parsedAddr, accAddr.String()) != 0 {
```

Comment:
interesting, how does `strings.Compare` differ from:
```go
if parsedAddr != accAddr.String() {
```

Reply:
> Compare is included only for symmetry with package bytes. It is usually clearer and always faster to use the built-in string comparison operators ==, <, >, and so on.

Originally I was going to use `bytes.Compare` with both `sdk.AccAddress`'s and ended up switching to `strings.Compare`.
I will update actually, I think the "built-in" comparison is a better option.